faker/providers/lorem/__init__.py

```diff
@@ -1,67 +1,108 @@
 localized = True
-default_locale = 'la'
+#default_locale is 'en_US' in the previous State of this application
+default_locale = 'la' 
+#external provider
+external_provider = ''
 
 from .. import BaseProvider
 
 
 class Provider(BaseProvider):
+    '''
+    factory.create will enter this class and will and will list all the
+    methods then will go to 'la.__init__'
+
+    Why does it goes to 'la' as soon as it finish to read this class
+    '''
+    #import pytest; pytest.set_trace()
     word_connector = ' '
     sentence_punctuation = '.'
     @classmethod
-    def word(cls):
+    def word(cls, ext_word_list=None):
         """
         Generate a random word
         :example 'lorem'
+
+        Keyword arguments:
+        'ext_word_list' --- a list of word you would like to have 
+                 instead of 'Lorem ipsum'
         """
+        if ext_word_list:
+            return cls.random_element(ext_word_list)
         return cls.random_element(cls.word_list)
 
     @classmethod
-    def words(cls, nb=3):
+    def words(cls, nb=3, ext_word_list=None):
         """
         Generate an array of random words
         :example array('Lorem', 'ipsum', 'dolor')
-        :param nb how many words to return
+
+        Keyword arguments:
+        nb --- how many words to return
+        'ext_word_list' --- a list of word you would like to have 
+                 instead of 'Lorem ipsum'
         """
+        if ext_word_list:
+            return [cls.word(ext_word_list) for _ in range(0, nb)]
         return [cls.word() for _ in range(0, nb)]
 
     @classmethod
-    def sentence(cls, nb_words=6, variable_nb_words=True):
+    def sentence(cls, nb_words=6, variable_nb_words=True, ext_word_list=None):
         """
         Generate a random sentence
         :example 'Lorem ipsum dolor sit amet.'
-        :param nb_words around how many words the sentence should contain
-        :param variable_nb_words set to false if you want exactly $nbWords returned,
+
+        Keyword arguments:
+        :param 'nb_words' --- around how many words the sentence should contain
+        :param 'variable_nb_words' --- set to false if you want exactly $nbWords returned,
             otherwise $nbWords may vary by +/-40% with a minimum of 1
+        :param 'ext_word_list' --- a list of word you would like to have 
+                 instead of 'Lorem ipsum'
         """
         if nb_words <= 0:
             return ''
 
         if variable_nb_words:
             nb_words = cls.randomize_nb_elements(nb_words)
 
-        words = cls.words(nb_words)
+        if ext_word_list:
+            words = cls.words(nb_words, ext_word_list)
+        else:
+            words = cls.words(nb_words)
```

Comment:
no need to check for `ext_word_list`. Just pass everything: `words = cls.words(nb=nb_words, ext_world_list=ext_world_list)`